store/src/executor/storage_handlers.rs

```diff
@@ -54,10 +57,34 @@ impl RequestHandler<TruncateTableAction> for ActionHandler {
             .get(tbl_name)
             .ok_or_else(|| ErrorCode::UnknownTable(format!("table not found: {:}", tbl_name)))?;
 
-        self.meta_node
-            .remove_table_data_parts(db_name, tbl_name)
-            .await;
+        let cr = LogEntry {
+            txid: None,
+            cmd: Cmd::TruncateTable {
+                db_name: db_name.clone(),
+                table_name: tbl_name.clone(),
+            },
+        };
 
-        Ok(TruncateTableResult {})
+        let rst = self
+            .meta_node
+            .write(cr)
+            .await
+            .map_err(|e| ErrorCode::MetaNodeInternalError(e.to_string()))?;
+
+        match rst {
+            AppliedState::DataPartsCount { prev, .. } => {
+                if let Some(prev) = prev {
+                    Ok(TruncateTableResult {
+                        trancated_table_data_parts_count: prev,
+                    })
+                } else {
+                    Err(ErrorCode::UnknownTable(format!(
+                        "table not found: {:}",
+                        tbl_name
+                    )))
+                }
```

Comment:
I supposed that `prev >= 0` and `result == 0` indicates a successful truncate.

`prev` being a `Some()`  does not imply some change to the table has taken place. There could be that there is such a table, but for some reason(no privilege?) truncating is canceled.

Reply:
ok,  that's more clearly